itwm_example/mco/mco.py

```diff
@@ -118,109 +77,114 @@ def optimize(self):
         return optimal_point, optimal_kpis
 
 
-def opt(weighted_score_func, initial_point, constraints):
-    """Partial func. Performs a scipy optimise with SLSQP method given the
-    scoring function, the initial point, and a set of constraints."""
-
-    return optimize.minimize(
-        weighted_score_func,
-        initial_point,
-        method="SLSQP",
-        bounds=constraints).x
-
-
-def get_weight_combinations(dimension, num_points, zero_values=True):
-    """Given the number of dimensions, this function provides all possible
-    combinations of weights adding to 1.0. For example, a dimension 3
-    will give all combinations (x, y, z) where x+y+z = 1.0.
-
-    The num_points parameter indicates how many divisions along a single
-    dimension will be performed. For example num_points == 3 will evaluate
-    for x being 0.0, 0.5 and 1.0. The returned (x, y, z) combinations will
-    of course be much higher than 3.
+class MCO(BaseMCO):
 
-    Note that if the zero_values parameter is set to false, then ensure
-    num_points > dimension in order for the generator to return any values.
+    optimizer = Type(IOptimizer)
+
+    def _optimizer_default(self):
+        return WeightedEvaluator
+
+    def get_scaling_factors(
+        self, evaluator, kpis, parameters, scaling_method=None
+    ):
+        """Calculates scaling factors for KPIs, defined in MCO.
+        Scaling factors are calculated (as required) by the provided scaling
+        method. In general, this provides normalization values for the possible
+        range of each KPI..
+        Performs scaling for all KPIs that have `auto_scale == True`.
+        Otherwise, keeps the default scale factor.
+
+        Parameters
+        ----------
+        evaluator: IOptimizer
+            Instance that provides optimization functionality
+        kpis: List[KPISpecification]
+            List of KPI objects to scale
+        scaling_method: callable
+            A method to scale KPI weights. Default set to the Sen's
+            "Multi-Objective Programming Method".
+        """
+        if scaling_method is None:
+            scaling_method = sen_scaling_method
+
+        evaluator = self.optimizer(evaluator, [1.0 for _ in kpis], parameters)
```

Comment:
Probably best to not overwrite the `evaluator` reference here, if only from a readability point of view. Maybe use `optimizer` or `weighted_optimizer` instead.